### PR TITLE
docs: add the release process

### PR DESCRIPTION
**Pull request by tbouffard**
It includes
  - code update
  - milestone and release notes management
  - npm package publish

### CODE REVIEW

File: packages/docs/release.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Release how-to
+
+This page explains the steps needed to release a new version of maxgraph@core.
+
+Currently, the release is entirely manual. Automation will come later based on the information provided here.
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
Currently, the release process is completely manual. Automation will come later based on the information provided here.
```

File: packages/docs/release.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+## Preparation
+
+Decide the new version depending on the types of changes:
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
Decide on the new version depending on the type of changes:
```

File: packages/docs/release.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- Follow [semver](https://semver.org/)
+- Until we release the first major version, bump the minor version when the release includes new features or breaking changes
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- Until we release the first major version, bump the minor version if the release contains new features or breaking changes.
```

File: packages/docs/release.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+Review the milestone related to the new release. **Note:** we always put issues related to a version in a Milestone whose
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
Check the milestone associated with the new release. **Note:** We always put issues related to a version into a Milestone whose
```

File: packages/docs/release.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+name matches the version.
+- Ensure the name of the milestone used for the new release version matches the name of the version that has is going to
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- Make sure that the name of the milestone used for the new release version matches the name of the version being
```

File: packages/docs/release.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+be released. Renamed it if needed.
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
released. Rename it if necessary.
```

File: packages/docs/release.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+- Check that all issues related to the upcoming release are attached to the milestone. In particular, check the issues with
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- Verify that all issues related to the upcoming release are attached to the milestone. In particular, check the issues that
```

File: packages/docs/release.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+[no milestone](https://github.com/maxGraph/maxGraph/issues?q=is%3Aissue+is%3Aclosed+no%3Amilestone).
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
[do not have a milestone](https://github.com/maxGraph/maxGraph/issues?q=is%3Aissue+is%3Aclosed+no%3Amilestone).
```

File: packages/docs/release.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+- Clean this opened milestone if some issues are still opened (move them to a new one or discard milestone from them)
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- Clean up this open milestone if some issues are still open (move them to a new milestone or discard the milestone from them).
```

File: packages/docs/release.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+- Close the milestone
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- Close the milestone.
```

File: packages/docs/release.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+Changes in the source code
+- Update the version in [package.json](../core/package.json) and the `VERSION` constant in the [Client](../core/src/Client.ts) file.
+- Update the [CHANGELOG](../../CHANGELOG.md) to list the major changes included in the new version. Be general and add a
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- Update the [CHANGELOG](../../CHANGELOG.md) to list the major changes included in the new version. Be generic and add a
```

File: packages/docs/release.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+link to the future GitHub release that will include detailed release notes like in the following.
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
link to the future GitHub release that will contain detailed release notes, as shown below.
```

File: packages/docs/release.md [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+```markdown
+For more details, see the [0.1.0 Changelog](https://github.com/maxGraph/maxGraph/releases/tag/v0.1.0) in the GitHub releases page.
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
For more details, see the [0.1.0 Changelog](https://github.com/maxGraph/maxGraph/releases/tag/v0.1.0) on the GitHub release page.
```

File: packages/docs/release.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+```
+- Do a single commit including the changes described above.
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- Make a single commit that includes the changes described above.
```

File: packages/docs/release.md [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+
+Create a git tag, prefixing the version by `v`. For instance, if the version is 0.2.0, run
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
Create a git tag, prefixing the version with a `v`. For example, if the version is 0.2.0, run
```

File: packages/docs/release.md [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+## Publish the npm package
+
+- Checkout the tag that has just been created
+- From packages/core, run `npm publish`
+
+
+## Create the GitHub release
+
+**Example**: use the [maxGraph 0.1.0 release](https://github.com/maxGraph/maxGraph/releases/tag/v0.1.0) for inspiration
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
**Example**: Use the [maxGraph 0.1.0 release](https://github.com/maxGraph/maxGraph/releases/tag/v0.1.0) for inspiration
```

File: packages/docs/release.md [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+Create a [new draft release](https://github.com/maxGraph/maxGraph/releases/)
+- name: use the version that has just been published
+- tag: use the tag create before
+- save it as a draft
+
+Generate the list of the important changes by using the [automatically generated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes).
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
Generate the list of the major changes by using the [automatically generated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes).
```

File: packages/docs/release.md [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+It relies on the labels of the merged Pull Requests included in this version and the [GitHub release configuration](../../.github/release.yml).
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
It is based on the labels of the merged Pull Requests included in this release and the [GitHub release configuration](../../.github/release.yml).
```

File: packages/docs/release.md [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+
+If the list is not accurate (for instance, an element is not placed in the right category), update the label(s) or the related
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
If the list is incorrect (for example, an item is not in the correct category), update the label(s) or the associated
```

File: packages/docs/release.md [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+
+Before you publish the release, make sure that a discussion will be created in the `Announces` category when the release
+is published.
+
+Publish the release.
+
+Review the newly created discussion in the [Announces](https://github.com/maxGraph/maxGraph/discussions/categories/announces) category:
+- adjust the title
+- pin the discussion and unpin the previous release announce
+- see the [maxGraph 0.1.0 release announce](https://github.com/maxGraph/maxGraph/discussions/147) for instance.
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
- see for example the [maxGraph 0.1.0 release announce](https://github.com/maxGraph/maxGraph/discussions/147).
```

File: packages/docs/release.md [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+
+## Update the integration examples repository
+
+Once done, create a Pull Request in the [integration examples repository](https://github.com/maxGraph/maxgraph-integration-examples)
+to use the new release. Use [PR #28](https://github.com/maxGraph/maxgraph-integration-examples/pull/28) as an example.
+
+**Note:** the examples will be updated automatically in the future, see the [issue #27](https://github.com/maxGraph/maxgraph-integration-examples/issues/27) for more details.
```
Comment (csouchet) [inline packages/docs/release.md]:
```suggestion
**Note:** The examples will be updated automatically in the future, see the [issue #27](https://github.com/maxGraph/maxgraph-integration-examples/issues/27) for more details.
```

File: packages/docs/release.md (hunk 1 above)
Comment (csouchet) [reply packages/docs/release.md]: Some changes proposed by Deepl.
You are free to add them or not 😉

**Comment by tbouffard** [comment]
@csouchet I have just integrated all your wording suggestions 😸

**Review by csouchet — APPROVED** [review]

**Pull request merged**